db/db_options_test.cc

```diff
@@ -383,6 +384,24 @@ TEST_F(DBOptionsTest, DeleteObsoleteFilesPeriodChange) {
   Close();
 }
 
+TEST_F(DBOptionsTest, MaxOpenFilesChange) {
+  SpecialEnv env(env_);
+  Options options;
+  options.max_open_files = -1;
+
+  Reopen(options);
+
+  LRUCache* tc = static_cast<LRUCache*>(dbfull()->TEST_table_cache());
```

Comment:
nit: Cache should have all the interfaces you need and you probably don't need to cast to LRUCache?